File: CS/WindowsApplication3/Custom LayoutControl/MyLayoutControl.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraLayout;

namespace DXSample {
    public class MyLayoutControl : LayoutControl
    {
        public MyLayoutControl() : base() { }
        public MyLayoutControl(bool createFast) : base(createFast) { }
        public MyLayoutControl(bool fAllowUseSplitters, bool fAllowUseTabbedGroup) : base(fAllowUseSplitters, fAllowUseTabbedGroup) { }

        public event CanHideItemEventHandler CanHideItem;

        internal void RaiseCanHideItem(BaseLayoutItem sender, CanHideItemEventArgs e)
        {
            if (CanHideItem != null)
                CanHideItem(sender, e);
        }

        protected override LayoutControlImplementor CreateILayoutControlImplementorCore()
        {
            return new MyLayoutControlImplementor(this);
        }
    }
}

File: VB/WindowsApplication3/Custom LayoutControl/MyLayoutControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraLayout

Namespace DXSample
	Public Class MyLayoutControl
		Inherits LayoutControl
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal createFast As Boolean)
			MyBase.New(createFast)
		End Sub
		Public Sub New(ByVal fAllowUseSplitters As Boolean, ByVal fAllowUseTabbedGroup As Boolean)
			MyBase.New(fAllowUseSplitters, fAllowUseTabbedGroup)
		End Sub

		Public Event CanHideItem As CanHideItemEventHandler

		Friend Sub RaiseCanHideItem(ByVal sender As BaseLayoutItem, ByVal e As CanHideItemEventArgs)
			RaiseEvent CanHideItem(sender, e)
		End Sub

		Protected Overrides Function CreateILayoutControlImplementorCore() As LayoutControlImplementor
			Return New MyLayoutControlImplementor(Me)
		End Function
	End Class
End Namespace